odoo/addons/base/models/ir_property.py

```diff
@@ -87,12 +85,43 @@ def _update_values(self, values):
 
     @api.multi
     def write(self, values):
-        return super(Property, self).write(self._update_values(values))
+        # if any of the records we're writing on has a res_id=False *or*
+        # we're writing a res_id=False on any record
+        default_set = False
+        if self._ids:
+            self.env.cr.execute(
+                'SELECT EXISTS (SELECT 1 FROM ir_property WHERE id in %s AND res_id IS NULL)', [self._ids])
```

Comment:
What about uses `LIMIT`?

Reply:
Doesn't matter, the select is not actually executed the DB engine will just use it as predicate when running exists.